### PR TITLE
Refresh DirectCSIVolume stats on volume updates

### PR DESCRIPTION
**Pull request by Praveenrajmani**
Also includes DirectCSIVolume type refactorings

### CODE REVIEW

File: pkg/apis/direct.csi.min.io/v1alpha1/types.go [hunk 1]
```diff
@@ -134,6 +134,13 @@ type DirectCSIVolume struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata"`
 
+	Status DirectCSIVolumeStatus `json:"status,omitempty"`
+	// +listType=atomic
+	// +optional
+	Conditions []metav1.Condition `json:"conditions"`
```
Comment (wlan0) [inline pkg/apis/direct.csi.min.io/v1alpha1/types.go:140]: Conditions should also go under status

File: Dockerfile [hunk 2]
```diff
@@ -1,11 +1,17 @@
+FROM golang:1.14 AS build
```
Comment (wlan0) [inline Dockerfile:1]: Do we need this change?

File: pkg/controller/volume_controller.go [hunk 3]
```diff
@@ -54,14 +56,36 @@ func (b *DirectCSIVolumeListener) Add(ctx context.Context, obj *v1alpha1.DirectC
 func (b *DirectCSIVolumeListener) Update(ctx context.Context, old, new *v1alpha1.DirectCSIVolume) error {
 	glog.V(1).Infof("Update called for DirectCSIVolume %s", new.ObjectMeta.Name)
 
+	directCSIClient := b.directcsiClient.DirectV1alpha1()
+	if utils.CheckVolumeStatusCondition(new.Conditions, "published", metav1.ConditionTrue) {
+		sc := utils.GetVolumeStatusCondition(new.Conditions, "volumestats")
+		duration := time.Since(sc.LastTransitionTime.Time)
+		if duration.Minutes() >= float64(5) {
```
Comment (wlan0) [inline pkg/controller/volume_controller.go:63]: if duration > 5 * time.Minute {

File: resources/workloads.yaml [hunk 4]
```diff
@@ -97,7 +97,7 @@ spec:
         - name: direct-csi
           image: $(REPOSITORY_ORG)/$(REPOSITORY_IMAGE):$(VERSION)
           args:
-            - "--identity=direct.csi.min.io"
```
Comment (wlan0) [inline resources/workloads.yaml:100]: Is this because plugin and kustomize yaml have diverged?

Comment (Praveenrajmani) [reply resources/workloads.yaml:100]: yes

**Review by wlan0 — APPROVED** [review]

**Pull request merged**